src/annotator/anchoring/pdf.js

```diff
@@ -72,11 +96,16 @@ function getPageTextContent(pageIndex) {
     return textContent;
   };
 
-  pageTextCache[pageIndex] = getPage(pageIndex)
-    .pdfPage.getTextContent({
+  // Fetch the text content for a given page as a string.
+  const getTextContent = async pageIndex => {
+    const pageView = await getPageView(pageIndex);
+    const textContent = await pageView.pdfPage.getTextContent({
       normalizeWhitespace: true,
-    })
-    .then(joinItems);
+    });
+    return joinItems(textContent.items);
+  };
```

Comment:
This step has been rewritten as an async function for readability, and to `await` the result of `getPageView(...)`.